source/common/http/header_utility.cc

```diff
@@ -41,12 +41,6 @@ HeaderUtility::HeaderData::HeaderData(const envoy::config::route::v3::HeaderMatc
     header_match_type_ = HeaderMatchType::Value;
     value_ = config.exact_match();
     break;
-  case envoy::config::route::v3::HeaderMatcher::HeaderMatchSpecifierCase::
```

Comment:
I don't love that there's a default in this switch statement.  lazy check: we'll fail to load before we can hit tihs case yeah?

Reply:
We could add LOG_EVERY_N error when the default is hit.

Reply:
This is intentional behavior it seems, see https://github.com/envoyproxy/envoy/blob/9785c3d08c30612a7161c9b7bba9772281f7c6f7/api/envoy/config/route/v3/route_components.proto#L1846